core/jobs/batch_jobs/exp_version_history_computation_job.py

```diff
@@ -0,0 +1,675 @@
+        the version number (if present).
+
+        Args:
+            change_list: list(ExplorationChange). The list of changes to check.
+
+        Returns:
+            Optional[int]. The revert version number (if present) or None.
+        """
+        for change in change_list:
+            if change.cmd == feconf.CMD_REVERT_COMMIT:
+                return change.version_number
+        return None
+
+    def _create_version_history_models(
+        self, model_group: FormattedModelGroupDict
+    ) -> Tuple[str, List[exp_models.ExplorationVersionHistoryModel]]:
+        """Creates the version history models for a particular exploration.
+
+        Args:
+            model_group: FormattedModelGroupDict. The formatted model group.
+
+        Returns:
+            Tuple[str, List[exp_models.ExplorationVersionHistoryModel]].
+            The exploration id along with the created version history models.
+        """
+        with datastore_services.get_ndb_context():
+            exp_v1 = model_group['exp_v1']
+            exp_vlatest = model_group['exp_vlatest']
+            commit_log_models = model_group['commit_log_models']
+            version_history_models = model_group['version_history_models']
+
+            exp_version = exp_vlatest.version
+            exp_id = exp_vlatest.id
+
+            versioned_explorations: List[exp_domain.Exploration] = (
+                [None] * exp_version
+            )
+            versioned_explorations[0] = exp_v1
+            versioned_explorations[exp_version - 1] = exp_vlatest
+
+            for version in range(1, exp_version + 1):
+                commit_log_model = commit_log_models[version - 1]
+                committer_id: str = commit_log_model.user_id
+                change_list: List[exp_domain.ExplorationChange] = []
+                for change_dict in commit_log_model.commit_cmds:
+                    change_list.append(exp_domain.ExplorationChange(
+                        change_dict
+                    ))
+
+                if version == 1:
+                    new_states_vh = {
+                        state_name: state_domain.StateVersionHistory(
+                            None, None, committer_id
+                        )
+                        for state_name in exp_v1.states
+                    }
+                    new_metadata_vh = exp_domain.MetadataVersionHistory(
+                        None, committer_id
+                    )
+                    new_committer_ids = [committer_id]
+                    new_vh_model = self._get_updated_version_history_model(
+                        version_history_models[version - 1],
+                        exp_id, version, committer_id,
+                        new_states_vh, new_metadata_vh, new_committer_ids
+                    )
+                    new_vh_model.update_timestamps()
+                    version_history_models[version - 1] = new_vh_model
```

Comment:
This can be done before this loop, right? The we don't need the if-else

Reply:
Removed it.